analyzers/src/SonarAnalyzer.Common/Helpers/CountComparisonKinds.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class CountComparisonKinds
+    {
+        public static bool AnyOrNotAny(this CountComparisonKind comparison) =>
+            comparison == CountComparisonKind.Any || comparison == CountComparisonKind.NotAny;
+
+        public static CountComparisonKind Count(this ComparisonKind comparison, int count) =>
+            comparison switch
+            {
+                ComparisonKind.Equals => Equals(count),
+                ComparisonKind.NotEquals => NotEquals(count),
+
+                ComparisonKind.GreaterThanOrEqual => GreaterThanOrEqual(count),
+                ComparisonKind.GreaterThan => GreaterThan(count),
+
+                ComparisonKind.LessThan => LessThan(count),
+                ComparisonKind.LessThanOrEqual => LessThanOrEqual(count),
+
+                _ => CountComparisonKind.None,
+            };
+
+        private static CountComparisonKind Equals(int count) =>
```

Comment:
Given the repetition, I thought this was the lesser of the evils to choose from. But if you think another one is better, be my guest.

Reply:
@pavel-mikula-sonarsource I think I processed all comments and/or gave a response.

Reply:
Let's postpone the decision. I'll keep this open for now and I'll take a look at this problem later, when the rest settles down.

Reply:
I agree.

Reply:
The API seems stable after the review I just done. To deal with this, let's introduce this
```
        private static CountComparisonResult Check(int count, CountComparisonResult smallerThanLimit, CountComparisonResult sameAsLimit, CountComparisonResult higherThanLimit, int limit = 0)
        {
            if(count < limit)
            {
                return smallerThanLimit;
            }
            else if(count > limit)
            {
                return higherThanLimit;
            }
            else
            {
                return sameAsLimit;
            }
        }
```
and remove all the intermediate methods. This is basically the pattern that all the intermediates do. I'm intentionally avoiding `lessThan`, `equals` and `greaterThan` names here to avoid confusion with the enum.

Reply:
I moved the limit parameter as It made the using code easier to read. I'm not really satisfied with this `Check` function, but the original was obviously not nice either, so, it's okay.